test/util.js

```diff
@@ -529,8 +542,19 @@ function checkCursorPaginatedHistoryObject ({
  * @param {String}   params.filters[i].prop
  * @param {Function} [params.filters[i].customGetValue(obj)]
  *   custom getter to value with prop (by default obj[prop])
- * @param {Function} [params.filters[i].customCheck(obj, value)]
- *   if specified, will perform a custom check on that property
+ *
+ * Custom check filter functions, one by filter type
+ * `getValue` is the getter function (obj[prop] or customGetValue(obj))
+ * @param {Function} [params.filters[i].customExactValueFilterCheck(obj, value)]
+ *  where value is a primitive type
+ *  default: (obj, value) => getValue(obj) === exactValue
+ * @param {Function} [params.filters[i].customArrayFilterCheck(obj, values)]
+ *  where values is an array
+ *  default: (obj, value) => values.includes(getValue(obj))
+ * @param {Function} [params.filters[i].customRangeFilterCheck(obj, rangeValues)]
+ *  where rangeValues is an object with the following properties: `gte` and `lte`
+ *  default: rangeValues.gte <= getValue(obj) && getValue(obj) <= rangeValues.lte
+ *
  * @param {Boolean}  [params.filters[i].isArrayFilter = false]
```

Comment:
Do you still need this now that you have three types of custom check functions, including a specific ones for array and range?

Reply:
Yes I still need `isArrayFilter` and `isRangeFilter` because some endpoints don't need custom check functions, so we need to differentiate them.

Reply:
Should not they be optional when providing corresponding custom check functions?

Reply:
Yes indeed, I make them optional if custom check functions are provided.